torch/_dynamo/variables/builtin.py

```diff
@@ -1677,11 +1677,21 @@ def _unimplemented():
         ):
             return ConstantVariable.create(op(left.value, right.value))
 
-        if (
-            (isinstance(left, StreamVariable) and isinstance(right, StreamVariable))
-            or (isinstance(left, EventVariable) and isinstance(right, EventVariable))
-        ) and op is operator.eq:
-            return ConstantVariable(op(left.value, right.value))
+        if isinstance(left, (StreamVariable, EventVariable)) or isinstance(
+            right, (StreamVariable, EventVariable)
+        ):
+            if (
+                (isinstance(left, StreamVariable) and isinstance(right, StreamVariable))
+                or (
+                    isinstance(left, EventVariable) and isinstance(right, EventVariable)
+                )
+            ) and op is operator.eq:
```

Comment:
nit: curious if this would work
```python
            if type(left) == type(right) and op is operator.eq:
```

Reply:
This is before my change actually. I guess this would work as well and looks cleaner.

Reply:
Better yet `type(left) is type(right)`?